### DISCUSSION TITLE
aioble - connect with linux pc

### DISCUSSION THREAD

**Discussion by newhinton** [micropython/micropython#10509] [Libraries & Drivers]
I am trying to facilitate communication between my KDE-System with an esp32. The documentation advises users to use the [aioble-library](https://github.com/micropython/micropython-lib/tree/master/micropython/bluetooth/aioble) which i tried.

The [temp-example](https://github.com/micropython/micropython-lib/blob/master/micropython/bluetooth/aioble/examples/temp_sensor.py) works insofar that i can see it with my bluetooth-scan (`bluetoothctl scan on`) but pairing fails. 

There is also no guide on what to do on the pc-side. I hope you can help me. Thanks!

**Comment by bixb922** [comment]
Have you tried to connect, without pairing? Should work. Pairing (with keys) is not supported on ESP32. It should also be fairly simple to write a client with Python and bleak or a similar BLE package acting as client/central.

**Reply by newhinton** [reply]
Yes, but no. bluetoothctr connects, but the immediately fails and shows it not beeing connected anymore.

```
[bluetooth]# connect XXXX
Attempting to connect to XXXX
[CHG] Device XXXX Connected: yes
Failed to connect: org.bluez.Error.Failed le-connection-abort-by-local
[CHG] Device XXXX Connected: no
```

**Reply by jimmo** [reply]
Pairing (&bonding) is supported on ESP32 in the nightly builds (and the upcoming v1.20, but not in v1.19).

**Reply by bixb922** [reply]
Good news about 1.20!

To test BLE connections there is also NRF connect for mobile phones, I have used that to test a BLE peripheral implemented with aioble, it is easy to use.
